src/ferry_planner/schedule.py

```diff
@@ -254,36 +308,180 @@ def _refresh_task(self) -> None:
             time.sleep(self.refresh_interval)
 
 
-def parse_schedule_html(*, html: str, date: datetime) -> Iterator[FerrySailing]:
+def parse_schedule_html(response: httpx.Response, date: datetime) -> HtmlParseResult:
+    result = HtmlParseResult()
```

Comment:
Better to create the object when returning, and keep the sailings as a separate variable until then